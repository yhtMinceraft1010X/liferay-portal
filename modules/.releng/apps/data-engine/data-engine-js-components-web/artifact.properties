artifact.git.id=b80ee6bc52c0532e058704f1fefbd2d16f6d6e2a
artifact.jsdoc.url=https://repository-cdn.liferay.com/nexus/content/repositories/liferay-public-releases/com/liferay/com.liferay.data.engine.js.components.web/1.0.60/com.liferay.data.engine.js.components.web-1.0.60-jsdoc.jar
artifact.sources-commercial.url=https://repository-cdn.liferay.com/nexus/content/repositories/liferay-public-releases/com/liferay/com.liferay.data.engine.js.components.web/1.0.60/com.liferay.data.engine.js.components.web-1.0.60-sources-commercial.jar
artifact.sources.url=https://repository-cdn.liferay.com/nexus/content/repositories/liferay-public-releases/com/liferay/com.liferay.data.engine.js.components.web/1.0.60/com.liferay.data.engine.js.components.web-1.0.60-sources.jar
artifact.url=https://repository-cdn.liferay.com/nexus/content/repositories/liferay-public-releases/com/liferay/com.liferay.data.engine.js.components.web/1.0.60/com.liferay.data.engine.js.components.web-1.0.60.jar