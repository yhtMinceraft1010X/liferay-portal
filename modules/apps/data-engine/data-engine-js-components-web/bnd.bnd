Bundle-Name: Liferay Data Engine JS Components Web
Bundle-SymbolicName: com.liferay.data.engine.js.components.web
Bundle-Version: 1.0.61
Web-ContextPath: /data-engine-js-components-web