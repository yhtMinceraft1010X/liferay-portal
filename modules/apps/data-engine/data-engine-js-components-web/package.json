{
	"dependencies": {
		"dom-align": "1.10.4",
		"moment": "^2.22.2"
	},
	"jest": {
		"collectCoverageFrom": [
			"src/main/resources/META-INF/resources/js/**/*.es.js"
		],
		"modulePathIgnorePatterns": [
			"/__mock__/",
			"/__utils__/"
		],
		"setupFiles": [
			"<rootDir>/jest-setup.config.js"
		],
		"snapshotSerializers": [
			"metal-jest-serializer"
		],
		"testPathIgnorePatterns": [
			"<rootDir>/test/js/[^/]+\\.js"
		]
	},
	"main": "js/index.es.js",
	"name": "data-engine-js-components-web",
	"private": true,
	"scripts": {
		"build": "liferay-npm-scripts build",
		"checkFormat": "liferay-npm-scripts check",
		"format": "liferay-npm-scripts fix",
		"test": "metalsoy -s \"test/**/*.soy\" -d \"test\" && liferay-npm-scripts test"
	},
	"version": "1.0.61"
}
