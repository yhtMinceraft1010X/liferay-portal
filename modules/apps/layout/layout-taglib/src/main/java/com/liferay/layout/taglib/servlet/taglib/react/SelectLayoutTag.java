/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

package com.liferay.layout.taglib.servlet.taglib.react;

import com.liferay.exportimport.kernel.staging.StagingUtil;
import com.liferay.item.selector.criteria.UUIDItemSelectorReturnType;
import com.liferay.layout.item.selector.LayoutItemSelectorReturnType;
import com.liferay.layout.taglib.internal.servlet.ServletContextUtil;
import com.liferay.portal.kernel.json.JSONArray;
import com.liferay.portal.kernel.json.JSONFactoryUtil;
import com.liferay.portal.kernel.json.JSONObject;
import com.liferay.portal.kernel.json.JSONUtil;
import com.liferay.portal.kernel.log.Log;
import com.liferay.portal.kernel.log.LogFactoryUtil;
import com.liferay.portal.kernel.model.Group;
import com.liferay.portal.kernel.model.Layout;
import com.liferay.portal.kernel.model.LayoutConstants;
import com.liferay.portal.kernel.security.auth.AuthTokenUtil;
import com.liferay.portal.kernel.service.LayoutLocalServiceUtil;
import com.liferay.portal.kernel.theme.ThemeDisplay;
import com.liferay.portal.kernel.util.ArrayUtil;
import com.liferay.portal.kernel.util.Constants;
import com.liferay.portal.kernel.util.HashMapBuilder;
import com.liferay.portal.kernel.util.HttpComponentsUtil;
import com.liferay.portal.kernel.util.ParamUtil;
import com.liferay.portal.kernel.util.PortalUtil;
import com.liferay.portal.kernel.util.WebKeys;
import com.liferay.taglib.util.IncludeTag;

import java.util.List;
import java.util.Map;
import java.util.Objects;

import javax.servlet.http.HttpServletRequest;
import javax.servlet.jsp.PageContext;

/**
 * @author Eudaldo Alonso
 * @author Marko Cikos
 */
public class SelectLayoutTag extends IncludeTag {

	public String getComponentId() {
		return _componentId;
	}

	public boolean getFollowURLOnTitleClick() {
		return _followURLOnTitleClick;
	}

	public String getItemSelectorReturnType() {
		return _itemSelectorReturnType;
	}

	public String getItemSelectorSaveEvent() {
		return _itemSelectorSaveEvent;
	}

	public String getNamespace() {
		return _namespace;
	}

	/**
	 * @deprecated As of Cavanaugh (7.4.x), with no direct replacement
	 */
	@Deprecated
	public String getPathThemeImages() {
		return _pathThemeImages;
	}

	/**
	 * @deprecated As of Cavanaugh (7.4.x), with no direct replacement
	 */
	@Deprecated
	public String getViewType() {
		return _viewType;
	}

	public boolean isCheckDisplayPage() {
		return _checkDisplayPage;
	}

	public boolean isEnableCurrentPage() {
		return _enableCurrentPage;
	}

	public boolean isMultiSelection() {
		return _multiSelection;
	}

	public boolean isPrivateLayout() {
		return _privateLayout;
	}

	public boolean isShowDraftLayouts() {
		return _showDraftLayouts;
	}

	public boolean isShowHiddenLayouts() {
		return _showHiddenLayouts;
	}

	public void setCheckDisplayPage(boolean checkDisplayPage) {
		_checkDisplayPage = checkDisplayPage;
	}

	public void setComponentId(String componentId) {
		_componentId = componentId;
	}

	public void setEnableCurrentPage(boolean enableCurrentPage) {
		_enableCurrentPage = enableCurrentPage;
	}

	public void setFollowURLOnTitleClick(boolean followURLOnTitleClick) {
		_followURLOnTitleClick = followURLOnTitleClick;
	}

	public void setItemSelectorReturnType(String itemSelectorReturnType) {
		_itemSelectorReturnType = itemSelectorReturnType;
	}

	public void setItemSelectorSaveEvent(String itemSelectorSaveEvent) {
		_itemSelectorSaveEvent = itemSelectorSaveEvent;
	}

	public void setMultiSelection(boolean multiSelection) {
		_multiSelection = multiSelection;
	}

	public void setNamespace(String namespace) {
		_namespace = namespace;
	}

	@Override
	public void setPageContext(PageContext pageContext) {
		super.setPageContext(pageContext);

		setServletContext(ServletContextUtil.getServletContext());
	}

	/**
	 * @deprecated As of Cavanaugh (7.4.x), with no direct replacement
	 */
	@Deprecated
	public void setPathThemeImages(String pathThemeImages) {
		_pathThemeImages = pathThemeImages;
	}

	public void setPrivateLayout(boolean privateLayout) {
		_privateLayout = privateLayout;
	}

	public void setShowDraftLayouts(boolean showDraftLayouts) {
		_showDraftLayouts = showDraftLayouts;
	}

	public void setShowHiddenLayouts(boolean showHiddenLayouts) {
		_showHiddenLayouts = showHiddenLayouts;
	}

	/**
	 * @deprecated As of Cavanaugh (7.4.x), with no direct replacement
	 */
	@Deprecated
	public void setViewType(String viewType) {
		_viewType = viewType;
	}

	@Override
	protected void cleanUp() {
		super.cleanUp();

		_checkDisplayPage = false;
		_componentId = null;
		_enableCurrentPage = false;
		_followURLOnTitleClick = false;
		_itemSelectorReturnType = null;
		_itemSelectorSaveEvent = null;
		_multiSelection = false;
		_namespace = null;
		_pathThemeImages = null;
		_privateLayout = false;
		_showDraftLayouts = false;
		_showHiddenLayouts = false;
		_viewType = null;
	}

	@Override
	protected String getPage() {
		return _PAGE;
	}

	@Override
	protected void setAttributes(HttpServletRequest httpServletRequest) {
		try {
			httpServletRequest.setAttribute(
				"liferay-layout:select-layout:data", _getData());
		}
		catch (Exception exception) {
			_log.error(exception);
		}
	}

	private Map<String, Object> _getData() throws Exception {
		String[] selectedLayoutIds = ParamUtil.getStringValues(
			getRequest(), "layoutUuid");

		return HashMapBuilder.<String, Object>put(
			"followURLOnTitleClick", _followURLOnTitleClick
		).put(
			"itemSelectorSaveEvent", _itemSelectorSaveEvent
		).put(
			"multiSelection", _multiSelection
		).put(
			"namespace", _namespace
		).put(
			"nodes", _getLayoutsJSONArray(selectedLayoutIds)
		).put(
			"selectedLayoutIds", selectedLayoutIds
		).build();
	}

	private JSONArray _getLayoutsJSONArray(
			long groupId, boolean privateLayout, long parentLayoutId,
			String[] selectedLayoutUuid)
		throws Exception {

		HttpServletRequest httpServletRequest = getRequest();

		ThemeDisplay themeDisplay =
			(ThemeDisplay)httpServletRequest.getAttribute(
				WebKeys.THEME_DISPLAY);

		JSONArray jsonArray = JSONFactoryUtil.createJSONArray();

		List<Layout> layouts = LayoutLocalServiceUtil.getLayouts(
			groupId, privateLayout, parentLayoutId);

		for (Layout layout : layouts) {
			if ((layout.isHidden() && !_showHiddenLayouts) ||
				_isExcludedLayout(layout) || StagingUtil.isIncomplete(layout)) {

				continue;
			}

			JSONObject jsonObject = JSONFactoryUtil.createJSONObject();

			JSONArray childrenJSONArray = _getLayoutsJSONArray(
				groupId, privateLayout, layout.getLayoutId(),
				selectedLayoutUuid);

			if (childrenJSONArray.length() > 0) {
				jsonObject.put("children", childrenJSONArray);
			}

			if ((_checkDisplayPage && !layout.isContentDisplayPage()) ||
				(_enableCurrentPage && (layout.getPlid() == _getSelPlid()))) {

				jsonObject.put("disabled", true);
			}

			jsonObject.put(
				"groupId", layout.getGroupId()
			).put(
				"icon", "page"
			).put(
				"id", layout.getUuid()
			).put(
				"layoutId", layout.getLayoutId()
			).put(
				"name", layout.getName(themeDisplay.getLocale())
			).put(
				"payload", _getPayload(layout, themeDisplay)
			).put(
				"privateLayout", layout.isPrivateLayout()
			).put(
				"returnType", getItemSelectorReturnType()
			).put(
				"url",
				PortalUtil.getLayoutRelativeURL(layout, themeDisplay, false)
			);

			if (ArrayUtil.contains(selectedLayoutUuid, layout.getUuid())) {
				jsonObject.put("selected", true);
			}

			jsonObject.put(
				"value", layout.getBreadcrumb(themeDisplay.getLocale()));

			jsonArray.put(jsonObject);
		}

		return jsonArray;
	}

	private JSONArray _getLayoutsJSONArray(String[] selectedLayoutIds)
		throws Exception {

		HttpServletRequest httpServletRequest = getRequest();

		ThemeDisplay themeDisplay =
			(ThemeDisplay)httpServletRequest.getAttribute(
				WebKeys.THEME_DISPLAY);

		Group group = themeDisplay.getScopeGroup();

		if ((_privateLayout && !group.hasPrivateLayouts()) ||
			(!_privateLayout && !group.hasPublicLayouts())) {

			return JSONFactoryUtil.createJSONArray();
		}

		return JSONUtil.put(
			JSONUtil.put(
				"children",
				_getLayoutsJSONArray(
					themeDisplay.getScopeGroupId(), _privateLayout, 0,
					selectedLayoutIds)
			).put(
				"disabled", true
			).put(
				"expanded", true
			).put(
				"icon", "home"
			).put(
				"id", "0"
			).put(
				"name", themeDisplay.getScopeGroupName()
			));
	}

	private String _getPayload(Layout layout, ThemeDisplay themeDisplay)
		throws Exception {

		if (Objects.equals(
				LayoutItemSelectorReturnType.class.getName(),
				getItemSelectorReturnType())) {

			return JSONUtil.put(
				"layoutId", layout.getLayoutId()
			).put(
				"name", layout.getName(themeDisplay.getLocale())
			).put(
				"plid", layout.getPlid()
			).put(
				"previewURL",
				() -> {
					String layoutURL = HttpComponentsUtil.addParameter(
						PortalUtil.getLayoutFullURL(layout, themeDisplay),
						"p_l_mode", Constants.PREVIEW);

					return HttpComponentsUtil.addParameter(
						layoutURL, "p_p_auth",
						AuthTokenUtil.getToken(getRequest()));
				}
			).put(
				"private", layout.isPrivateLayout()
			).put(
				"url", PortalUtil.getLayoutFullURL(layout, themeDisplay)
			).put(
				"uuid", layout.getUuid()
			).toString();
		}
		else if (Objects.equals(
					UUIDItemSelectorReturnType.class.getName(),
					getItemSelectorReturnType())) {

			return layout.getUuid();
		}

		return PortalUtil.getLayoutRelativeURL(layout, themeDisplay, false);
	}

	private long _getSelPlid() {
		return ParamUtil.getLong(
			getRequest(), "selPlid", LayoutConstants.DEFAULT_PLID);
	}

	private boolean _isExcludedLayout(Layout layout) {
		if (!layout.isTypeContent()) {
			return false;
		}

		if (layout.fetchDraftLayout() != null) {
			if (_showDraftLayouts) {
				return false;
			}

			return !layout.isPublished();
		}

		if (layout.isApproved() && !layout.isHidden() && !layout.isSystem()) {
			return false;
		}

		return true;
	}

	private static final String _PAGE = "/select_layout/page.jsp";

	private static final Log _log = LogFactoryUtil.getLog(
		SelectLayoutTag.class);

	private boolean _checkDisplayPage;
	private String _componentId;
	private boolean _enableCurrentPage;
	private boolean _followURLOnTitleClick;
	private String _itemSelectorReturnType;
	private String _itemSelectorSaveEvent;
	private boolean _multiSelection;
	private String _namespace;
	private String _pathThemeImages;
	private boolean _privateLayout;
	private boolean _showDraftLayouts;
	private boolean _showHiddenLayouts;
	private String _viewType;

}