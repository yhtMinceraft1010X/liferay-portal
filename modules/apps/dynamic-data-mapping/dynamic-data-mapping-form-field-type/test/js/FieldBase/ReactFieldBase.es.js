/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

import {act, fireEvent, render} from '@testing-library/react';
import {PageProvider} from 'data-engine-js-components-web';
import React from 'react';

import '@testing-library/jest-dom/extend-expect';

import {FieldBase} from '../../../src/main/resources/META-INF/resources/FieldBase/ReactFieldBase.es';

const spritemap = 'icons.svg';

const FieldBaseWithProvider = (props) => (
	<PageProvider value={{editingLanguageId: 'en_US'}}>
		<FieldBase {...props} />
	</PageProvider>
);

describe('ReactFieldBase', () => {
	// eslint-disable-next-line no-console
	const originalWarn = console.warn;

	beforeAll(() => {
		window.themeDisplay = {
			...window.themeDisplay,
			getPathThemeImages: () => 'http://localhost:8080',
		};

		// eslint-disable-next-line no-console
		console.warn = (...args) => {
			if (/DataProvider: Trying/.test(args[0])) {
				return;
			}
			originalWarn.call(console, ...args);
		};
	});

	afterAll(() => {
		// eslint-disable-next-line no-console
		console.warn = originalWarn;
	});

	beforeEach(() => {
		jest.useFakeTimers();
		fetch.mockResponseOnce(JSON.stringify({}));
	});

	it('renders the default markup', () => {
		const {container} = render(
			<FieldBaseWithProvider spritemap={spritemap} />
		);

		act(() => {
			jest.runAllTimers();
		});

		expect(container).toMatchSnapshot();
	});

	it('renders the FieldBase with required', () => {
		const {container} = render(
			<FieldBaseWithProvider required spritemap={spritemap} />
		);

		act(() => {
			jest.runAllTimers();
		});

		expect(container).toMatchSnapshot();
	});

	it('renders the FieldBase with id', () => {
		const {container} = render(
			<FieldBaseWithProvider id="Id" spritemap={spritemap} />
		);

		act(() => {
			jest.runAllTimers();
		});

		expect(container).toMatchSnapshot();
	});

	it('renders the FieldBase with help text', () => {
		const {container} = render(
			<FieldBaseWithProvider
				spritemap={spritemap}
				tip="Type something!"
			/>
		);

		act(() => {
			jest.runAllTimers();
		});

		expect(container).toMatchSnapshot();
	});

	it('renders the FieldBase with label', () => {
		const {container} = render(
			<FieldBaseWithProvider label="Text" spritemap={spritemap} />
		);

		act(() => {
			jest.runAllTimers();
		});

		expect(container).toMatchSnapshot();
	});

	it('renders the FieldBase with tooltip', () => {
		const {container} = render(
			<FieldBaseWithProvider spritemap={spritemap} tooltip="Tooltip" />
		);

		act(() => {
			jest.runAllTimers();
		});

		expect(container.querySelector('.ddm-tooltip')).not.toBeNull();
	});

	it('does not render the label if showLabel is false', () => {
		const {container} = render(
			<FieldBaseWithProvider
				label="Text"
				showLabel={false}
				spritemap={spritemap}
			/>
		);

		act(() => {
			jest.runAllTimers();
		});

		expect(container).toMatchSnapshot();
	});

	it('renders the FieldBase with contentRenderer', () => {
		const {container} = render(
			<FieldBaseWithProvider spritemap={spritemap}>
				<div>
					<h1>Foo bar</h1>
				</div>
			</FieldBaseWithProvider>
		);

		act(() => {
			jest.runAllTimers();
		});

		expect(container).toMatchSnapshot();
	});

	it('renders the add button when repeatable is true', () => {
		const {container} = render(
			<FieldBaseWithProvider
				label="Text"
				repeatable={true}
				showLabel={false}
				spritemap={spritemap}
			/>
		);
		act(() => {
			jest.runAllTimers();
		});

		expect(container).toMatchSnapshot();
	});

	it('does not render the add button when repeatable is true and the maximum limit of repetions is reached', () => {
		const {container} = render(
			<FieldBaseWithProvider
				label="Text"
				overMaximumRepetitionsLimit={true}
				repeatable={true}
				showLabel={false}
				spritemap={spritemap}
			/>
		);
		act(() => {
			jest.runAllTimers();
		});

		expect(container).toMatchSnapshot();
	});

	it('shows the popover for Format field when hovering over the tooltip icon', async () => {
		const {container, findByTestId, getByRole, getByText} = render(
			<FieldBaseWithProvider
				fieldName="inputMaskFormat"
				spritemap={spritemap}
				tooltip="Tooltip Description"
			/>
		);

		const tooltipIcon = container.querySelector('.ddm-tooltip');

		fireEvent.mouseOver(tooltipIcon);

		const clayPopover = await findByTestId('clayPopover');

		expect(clayPopover.style).toHaveProperty('maxWidth', '256px');

		expect(getByRole('img')).toHaveAttribute('height', '170');
		expect(getByRole('img')).toHaveAttribute(
			'src',
			'http://localhost:8080/forms/input_mask_format.png'
		);
		expect(getByRole('img')).toHaveAttribute('width', '232');

		expect(getByText('input-mask-format')).toBeInTheDocument();
		expect(getByText('Tooltip Description')).toBeInTheDocument();
	});

	describe('Hide Field', () => {
		it('renders the FieldBase with hideField markup', () => {
			const {getAllByText, getByText} = render(
				<FieldBaseWithProvider
					hideField
					label="Text"
					spritemap={spritemap}
				/>
			);

			expect(getByText('hidden')).toBeInTheDocument();

			const allByText = getAllByText('Text');
			expect(allByText).toHaveLength(2);
			expect(allByText[0]).toBeInTheDocument();
			expect(allByText[1]).toBeInTheDocument();

			expect(getByText('hidden').parentNode).toHaveAttribute(
				'class',
				'label ml-1 label-secondary'
			);
			expect(allByText[0]).toHaveAttribute('class', 'text-secondary');
			expect(allByText[1]).toHaveAttribute('class', 'sr-only');
		});

		it('renders the FieldBase with hideField markup when the label is empty', () => {
			const {getByText} = render(
				<FieldBaseWithProvider
					hideField
					label=""
					spritemap={spritemap}
				/>
			);

			expect(getByText('hidden')).toBeInTheDocument();

			expect(getByText('hidden').parentNode).toHaveAttribute(
				'class',
				'label ml-1 label-secondary'
			);
		});
	});
});
