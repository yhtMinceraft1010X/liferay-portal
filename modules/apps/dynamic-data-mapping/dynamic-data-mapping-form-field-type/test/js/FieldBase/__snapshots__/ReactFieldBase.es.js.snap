// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReactFieldBase does not render the add button when repeatable is true and the maximum limit of repetions is reached 1`] = `
<div>
  <div
    class="form-group hide"
  >
    <div
      class="lfr-ddm-form-field-repeatable-toolbar"
    >
      <button
        aria-label="add-duplicate-field"
        class="ddm-form-field-repeatable-add-button p-0 hide btn btn-sm btn-primary"
        title="duplicate"
        type="button"
      >
        <svg
          class="lexicon-icon lexicon-icon-plus"
          role="presentation"
        >
          <use
            xlink:href="#plus"
          />
        </svg>
      </button>
    </div>
    <label
      aria-labelledby="undefined_fieldDetails"
      class="ddm-empty"
      tabindex="0"
    />
    <input
      name="undefined_edited"
      type="hidden"
      value="false"
    />
    <span
      class="sr-only"
      id="undefined_fieldDetails"
    >
      Text
    </span>
  </div>
</div>
`;

exports[`ReactFieldBase does not render the label if showLabel is false 1`] = `
<div>
  <div
    aria-labelledby="undefined_fieldDetails"
    class="form-group hide"
    tabindex="0"
  >
    <input
      name="undefined_edited"
      type="hidden"
      value="false"
    />
    <span
      class="sr-only"
      id="undefined_fieldDetails"
    >
      Text
    </span>
  </div>
</div>
`;

exports[`ReactFieldBase renders the FieldBase with contentRenderer 1`] = `
<div>
  <div
    aria-labelledby="undefined_fieldDetails"
    class="form-group hide"
    tabindex="0"
  >
    <div>
      <h1>
        Foo bar
      </h1>
    </div>
    <input
      name="undefined_edited"
      type="hidden"
      value="false"
    />
    
  </div>
</div>
`;

exports[`ReactFieldBase renders the FieldBase with help text 1`] = `
<div>
  <div
    aria-labelledby="undefined_fieldDetails"
    class="form-group hide"
    tabindex="0"
  >
    <input
      name="undefined_edited"
      type="hidden"
      value="false"
    />
    <div
      aria-hidden="true"
      class="form-feedback-group lfr-de__field-feedback"
    >
      <div>
        Type something!
      </div>
    </div>
    <span
      class="sr-only"
      id="undefined_fieldDetails"
    >
      Type something!
    </span>
  </div>
</div>
`;

exports[`ReactFieldBase renders the FieldBase with id 1`] = `
<div>
  <div
    aria-labelledby="Id_fieldDetails"
    class="form-group hide"
    tabindex="0"
  >
    <input
      name="undefined_edited"
      type="hidden"
      value="false"
    />
    
  </div>
</div>
`;

exports[`ReactFieldBase renders the FieldBase with label 1`] = `
<div>
  <div
    class="form-group hide"
  >
    <label
      aria-labelledby="undefined_fieldDetails"
      class="ddm-label"
      tabindex="0"
    >
      Text
    </label>
    <input
      name="undefined_edited"
      type="hidden"
      value="false"
    />
    <span
      class="sr-only"
      id="undefined_fieldDetails"
    >
      Text
    </span>
  </div>
</div>
`;

exports[`ReactFieldBase renders the FieldBase with required 1`] = `
<div>
  <div
    class="form-group hide"
  >
    <label
      aria-labelledby="undefined_fieldDetails"
      class="ddm-label"
      tabindex="0"
    >
      <span
        class="ddm-label-required reference-mark"
      >
        <svg
          class="lexicon-icon lexicon-icon-asterisk"
          role="presentation"
        >
          <use
            xlink:href="#asterisk"
          />
        </svg>
      </span>
    </label>
    <input
      name="undefined_edited"
      type="hidden"
      value="false"
    />
    <span
      class="sr-only"
      id="undefined_fieldDetails"
    >
      required
    </span>
  </div>
</div>
`;

exports[`ReactFieldBase renders the add button when repeatable is true 1`] = `
<div>
  <div
    class="form-group hide"
  >
    <div
      class="lfr-ddm-form-field-repeatable-toolbar"
    >
      <button
        aria-label="add-duplicate-field"
        class="ddm-form-field-repeatable-add-button p-0 btn btn-sm btn-primary"
        title="duplicate"
        type="button"
      >
        <svg
          class="lexicon-icon lexicon-icon-plus"
          role="presentation"
        >
          <use
            xlink:href="#plus"
          />
        </svg>
      </button>
    </div>
    <label
      aria-labelledby="undefined_fieldDetails"
      class="ddm-empty"
      tabindex="0"
    />
    <input
      name="undefined_edited"
      type="hidden"
      value="false"
    />
    <span
      class="sr-only"
      id="undefined_fieldDetails"
    >
      Text
    </span>
  </div>
</div>
`;

exports[`ReactFieldBase renders the default markup 1`] = `
<div>
  <div
    aria-labelledby="undefined_fieldDetails"
    class="form-group hide"
    tabindex="0"
  >
    <input
      name="undefined_edited"
      type="hidden"
      value="false"
    />
    
  </div>
</div>
`;
